### PR TITLE
test: cleanup onchain bats tests

### PR DESCRIPTION
**Pull request by vindard**

### CODE REVIEW

File: test/bats/onchain-payments.bats [hunk 1]
```diff
@@ -15,6 +15,10 @@ teardown_file() {
   stop_exporter
 }
 
+teardown() {
+  [ "$(balance_for_check)" = 0 ]
```
Comment (bodymindarts) [inline test/bats/onchain-payments.bats:19]: `[[ "$(balance_for_check)" = "0" ]] || exit 1`

File: test/bats/onchain-payments.bats [hunk 2]
```diff
@@ -15,6 +15,10 @@ teardown_file() {
...
+}
+
 @test "onchain payments: setup user" {
   input=$(
     jq -n \
```
Comment (vindard) [inline test/bats/onchain-payments.bats:24]: Remove `-n`

File: test/bats/onchain-payments.bats [hunk 3]
```diff
@@ -48,9 +54,14 @@ teardown_file() {
 
   bitcoin_cli sendtoaddress "$address" 0.01
   bitcoin_cli -generate 2
-  sleep 5
-}
 
+  check_for_settled() {
```
Comment (vindard) [inline test/bats/onchain-payments.bats:58]: Move to helpers

File: test/bats/onchain-payments.bats [hunk 4]
```diff
@@ -66,20 +77,20 @@ teardown_file() {
...
-  exec_graphql 'alice' 'transactions' '{"first":1}'
-  txid="$(graphql_output '.data.me.defaultAccount.transactions.edges[0].node.settlementVia.transactionHash')"
-  [ "${txid}" != "null" ]
-  bitcoin_cli -generate 2
-  sleep 5 # wait for settle
 
-  exec_graphql 'alice' 'transactions' '{"first":1}'
-  settled_status="$(graphql_output '.data.me.defaultAccount.transactions.edges[0].node.status')"
-  [ "${settled_status}" = "SUCCESS" ]
+  check_for_broadcast() {
```
Comment (vindard) [inline test/bats/onchain-payments.bats:81]: Move to helpers

**Review by nicolasburtey — APPROVED** [review]
approving just in case we want to merge, but haven't reviewed the code

**Pull request merged**